File: packages/ios/Sources/Components/Chip/Chip.figma.swift
import SwiftUI
import Figma

struct KozmosChipConnect: FigmaConnect {
    let component = KozmosChip<EmptyView>.self
    let figmaNodeUrl = "https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=227-1329"

    @FigmaString("Label Text")
    var label: String = "Chip"

    @FigmaEnum(
        "Variant",
        mapping: [
            "Neutral": KozmosChipVariant.neutral,
            "Brand": KozmosChipVariant.brand,
            "Destructive": KozmosChipVariant.destructive
        ]
    )
    var chipVariant: KozmosChipVariant = .neutral

    @FigmaEnum(
        "Size",
        mapping: [
            "Small": KozmosChipSize.sm,
            "Default": KozmosChipSize.`default`,
            "Large": KozmosChipSize.lg
        ]
    )
    var size: KozmosChipSize = .default

    @FigmaEnum(
        "State",
        mapping: [
            "Default": "default",
            "Selected": "selected",
            "Disabled": "disabled"
        ]
    )
    var state: String = "default"

    @FigmaEnum(
        "Removable",
        mapping: [
            "False": false,
            "True": true
        ]
    )
    var removable: Bool = false

    var body: some View {
        KozmosChip(
            text: self.label,
            variant: self.chipVariant,
            size: self.size,
            selected: self.state == "selected",
            disabled: self.state == "disabled",
            onRemove: self.removable ? {} : nil
        )
    }
}

File: packages/android/src/main/java/com/kozmos/components/Chip/Chip.figma.kt
package com.kozmos.components.chip

import androidx.compose.runtime.Composable
import com.figma.code.connect.Figma
import com.figma.code.connect.FigmaConnect
import com.figma.code.connect.FigmaProperty
import com.figma.code.connect.FigmaType

@FigmaConnect("https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=227-1329")
class KozmosChipConnect {
    @FigmaProperty(FigmaType.Text, "Label Text")
    val label: String = "Chip"

    @FigmaProperty(FigmaType.Enum, "Variant")
    val variant: ChipVariant = Figma.mapping(
        "Neutral" to ChipVariant.Neutral,
        "Brand" to ChipVariant.Brand,
        "Destructive" to ChipVariant.Destructive
    )

    @FigmaProperty(FigmaType.Enum, "Size")
    val size: ChipSize = Figma.mapping(
        "Small" to ChipSize.Sm,
        "Default" to ChipSize.Default,
        "Large" to ChipSize.Lg
    )

    @FigmaProperty(FigmaType.Enum, "State")
    val state: String = Figma.mapping(
        "Default" to "default",
        "Selected" to "selected",
        "Disabled" to "disabled"
    )

    @FigmaProperty(FigmaType.Enum, "Removable")
    val removable: Boolean = Figma.mapping(
        "False" to false,
        "True" to true
    )

    @Composable
    fun ComponentExample() {
        KozmosChip(
            text = label,
            variant = variant,
            size = size,
            selected = state == "selected",
            enabled = state != "disabled",
            onRemove = if (removable) ({}) else null
        )
    }
}
